addOns/accessControl/src/main/java/org/zaproxy/zap/extension/accessControl/AccessControlAPI.java

```diff
@@ -0,0 +1,217 @@
+
+                startOptions.targetContext = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID);
+
+                if (usersExtension == null) {
+                    usersExtension =
+                            Control.getSingleton()
+                                    .getExtensionLoader()
+                                    .getExtension(ExtensionUserManagement.class);
+                }
+
+                List<User> users = new ArrayList<User>();
+
+                String[] commaSeparatedUserIDs =
+                        ApiUtils.getNonEmptyStringParam(params, PARAM_USER_ID).split("\\s*,\\s*");
+
+                for (int i = 0; i < commaSeparatedUserIDs.length; i++) {
+                    int userID = Integer.parseInt(commaSeparatedUserIDs[i]);
+                    users.add(
+                            usersExtension
+                                    .getContextUserAuthManager(
+                                            startOptions.targetContext.getIndex())
+                                    .getUserById(userID));
+                }
+
+                startOptions.targetUsers = users;
+
+                // Add unauthenticated user
+                if (params.has(PARAM_UNAUTH_USER)
+                        && StringUtils.isNotBlank(params.getString(PARAM_UNAUTH_USER))
+                        && params.getBoolean(PARAM_UNAUTH_USER)) {
+                    startOptions.targetUsers.add(null);
+                }
+
+                if (params.has(PARAM_RAISE_ALERT)
+                        && StringUtils.isNotBlank(params.getString(PARAM_RAISE_ALERT))) {
+                    startOptions.raiseAlerts = params.getBoolean(PARAM_RAISE_ALERT);
+                } else {
+                    startOptions.raiseAlerts = true;
+                }
+
+                if (params.has(PARAM_ALERT_RISK_LEVEL)
+                        && StringUtils.isNotBlank(params.getString(PARAM_ALERT_RISK_LEVEL))) {
+                    startOptions.alertRiskLevel = params.getInt(PARAM_ALERT_RISK_LEVEL);
+                } else {
+                    startOptions.alertRiskLevel = org.parosproxy.paros.core.scanner.Alert.RISK_HIGH;
+                }
+
+                extension.startScan(startOptions);
+
+                result = ApiResponseElement.OK;
+                break;
+
+            case ACTION_WRITE_HTML_REPORT:
+                LOGGER.debug("Write HTML report called");
+
+                // The path for the XSL file
+                File xslFile =
+                        new File(
+                                Constant.getZapHome(),
+                                "xml" + File.separator + "reportAccessControl.html.xsl");
+
+                File reportFile = new File(params.getString(PARAM_FILENAME));
+
+                try {
+                    ReportGenerator.XMLToHtml(
+                            extension.generateLastScanXMLReport(params.getInt(PARAM_CONTEXT_ID)),
```

Comment:
- Why not call `generateAccessControlReport`?
 - Should use `ApiUtils.getIntParam(params, PARAM_CONTEXT_ID)` which has proper error handling.